lnwallet/chanfunding/psbt_assembler.go

```diff
@@ -0,0 +1,372 @@
+	// only closed exactly once.
+	signalPsbtReady sync.Once
+}
+
+// BindKeys sets both the remote and local node's keys that will be used for the
+// channel funding multisig output.
+func (i *PsbtIntent) BindKeys(localKey *keychain.KeyDescriptor,
+	remoteKey *btcec.PublicKey) {
+
+	i.localKey = localKey
+	i.remoteKey = remoteKey
+	i.State = PsbtOutputKnown
+}
+
+// FundingParams returns the parameters that are necessary to start funding the
+// channel output this intent was created for. It returns the bech32 encoded
+// P2WSH funding address, the exact funding amount and a raw PSBT that contains
+// exactly one output that encodes the previous two parameters.
+func (i *PsbtIntent) FundingParams(netParams *chaincfg.Params) (string, int64,
+	[]byte, error) {
+
+	// The funding output needs to be known already at this point, which
+	// means we need to have the local and remote multisig keys bound
+	// already.
+	witnessScript, out, err := i.FundingOutput()
+	if err != nil {
+		return "", 0, nil, fmt.Errorf("unable to create funding "+
+			"output: %v", err)
+	}
+	witnessScriptHash := sha256.Sum256(witnessScript)
+
+	// Encode the address in the human readable bech32 format.
+	addr, err := btcutil.NewAddressWitnessScriptHash(
+		witnessScriptHash[:], netParams,
+	)
+	if err != nil {
+		return "", 0, nil, fmt.Errorf("unable to encode address: %v",
+			err)
+	}
+
+	// We'll also encode the address/amount in a machine readable raw PSBT
+	// format. If the user supplied a base PSBT, we'll add the output to
+	// that one, otherwise we'll create a new one.
+	packet := i.BasePsbt
+	if packet == nil {
+		packet, err = psbt.New(nil, nil, 2, 0, nil)
+		if err != nil {
+			return "", 0, nil, fmt.Errorf("unable to create PSBT: "+
+				"%v", err)
+		}
+	}
+	packet.UnsignedTx.TxOut = append(packet.UnsignedTx.TxOut, out)
+	packet.Outputs = append(packet.Outputs, psbt.POutput{})
+
+	// Serialize the new or updated PSBT into the raw binary format.
+	var buf bytes.Buffer
+	err = packet.Serialize(&buf)
+	if err != nil {
+		return "", 0, nil, fmt.Errorf("unable to serialize PSBT: %v",
+			err)
+	}
+
+	return addr.EncodeAddress(), out.Value, buf.Bytes(), nil
+}
+
+// Verify makes sure the PSBT that is given to the intent has an output that
+// sends to the channel funding multisig address with the correct amount. A
+// simple check that there is a large enough input present is also performed.
+func (i *PsbtIntent) Verify(rawPsbt []byte) error {
+	// We want the PSBT in the raw binary format, base64 is only used when
+	// a direct user interaction is involved.
+	packet, err := psbt.NewFromRawBytes(bytes.NewReader(rawPsbt), false)
+	if err != nil {
+		return fmt.Errorf("error parseing PSBT: %v", err)
+	}
+
+	// Try to locate the channel funding multisig output.
+	_, expectedOutput, err := i.FundingOutput()
+	if err != nil {
+		return fmt.Errorf("funding output cannot be created: %v", err)
+	}
+	outputFound := false
+	for _, out := range packet.UnsignedTx.TxOut {
```

Comment:
We should also check for the proper output value as well.

Reply:
That's why I'm using `DeepEqual` so both the output value and PK script are checked. Or what am I missing?